src/docs/release/breaking-changes/layout-builder-optimization.md

```diff
@@ -0,0 +1,170 @@
+The app may exhibit some combination of the following symptoms:
+
+* The `builder` function is not called when it would before the upgrade to the
+  Flutter version that introduced the optimization.
+* The UI of a widget is missing.
+* The UI of a widget is not updating.
+
+## Description of change
+
+The pull request introducing the change can be found [here][6]. Prior to the
+optimization the builder function passed to `LayoutBuilder` or
+`SliverLayoutBuilder` would be called when any one of the following happens:
+
+1. `LayoutBuilder` is rebuilt due to widget configuration change (this typically
+   happens when the widget that uses `LayoutBuilder` rebuilds due to `setState`,
+   `didUpdateWidget` or `didChangeDependencies`).
+1. `LayoutBuilder` is laid out and receives layout constraints from its parent
+   that are _different_ from the last received constraints.
+1. `LayoutBuilder` is laid out and receives layout constraints from its parent
+   that are the _same_ as the constraints received last time.
+
+After the optimization the builder function is no longer called in the latter
+case. If the constraints are the same and the widget configuration did not
+change, the builder function is not called.
+
+Your app can break if it relies on the relayout to cause the rebuilding of the
+`LayoutBuilder` rather than on an explicit call to `setState`. This usually
+happens by accident. You meant to add `setState`, but you forgot because the app
+continued functioning as you wanted, and therefore nothing reminded you to add
+it.
+
+## Migration guide
+
+Look for usages of `LayoutBuilder` and `SliverLayoutBuilder` and make sure to
+call `setState` any time the widget state changes.
+
+**Example**: in the example below the contents of the builder function depend
+on the value of the `_counter` field. Therefore, whenever the value is updated,
+you should call `setState` to tell the framework to rebuild the widget. However,
+this example may have previously worked even without calling `setState`, if the
+`_ResizingBox` triggers a relayout of `LayoutBuilder` (see the implementation
+of `_ResizingBox` in the [design doc][1]).
+
+Code before migration (note the missing `setState` inside the `onPressed`
+callback):
+
+```dart
+class Counter extends StatefulWidget {
+  Counter({Key key}) : super(key: key);
+
+  @override
+  _CounterState createState() => _CounterState();
+}
+
+class _CounterState extends State<Counter> {
+  int _counter = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(child: Container(
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return _ResizingBox(
+            FlatButton(onPressed: () {
+              _counter++;
+            }, child: Text('Increment Counter')),
+            Text(_counter.toString()),
+          );
+        },
+      ),
+    ));
+  }
+}
+```
+
+Code after migration (`setState` added to `onPressed`):
+
```

Comment:
Same comment as above. If the API is not in the current stable release (which is what Travis tests against), OR if it's not a complete example that can pass analysis on its own, add the <!-- skip --> line.

Reply:
Done (fixed the code).